include/etcpal/cpp/uuid.h

```diff
@@ -0,0 +1,210 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************
+ * This file is a part of EtcPal. For more information, go to:
+ * https://github.com/ETCLabs/EtcPal
+ ******************************************************************************/
+
+/// \file etcpal/cpp/uuid.h
+/// \brief C++ wrapper and utilities for etcpal/uuid.h
+
+#ifndef ETCPAL_CPP_UUID_H_
+#define ETCPAL_CPP_UUID_H_
+
+#include <cstdint>
+#include <cstring>
+#include <array>
+#include <string>
+#include "etcpal/uuid.h"
+
+namespace etcpal
+{
+/// \defgroup etcpal_cpp_uuid etcpal_cpp_uuid
+/// \ingroup etcpal_cpp
+/// \brief C++ utilities for the \ref etcpal_uuid module.
+
+/// \ingroup etcpal_cpp_uuid
+/// \brief A wrapper class for the EtcPal UUID type.
+///
+/// Provides C++ syntactic sugar for working with UUIDs. UUIDs are considered immutable after they
+/// are created, so there is no non-const access to the underlying data as of now.
+class Uuid
+{
+public:
+  /// \brief Constructs a null UUID by default.
+  Uuid() = default;
+  Uuid(const EtcPalUuid& c_uuid);
```

Comment:
Should this be marked explicit?

Reply:
I would also like to allow implicit conversions here:
```C++
void DoSomething(const EtcPalUuid& c_uuid)
{
  etcpal::Uuid uuid = c_uuid;
  ...
```

If you agree, I'll add a comment like you asked for on `Result` here as well.

Reply:
Oh, right... It's good to see that explicit is starting to be ingrained in my head. :-) Yes, please add a comment.